lib/manageiq_performance/middlewares/memory.rb

```diff
@@ -0,0 +1,55 @@
+module ManageIQPerformance
+  module Middlewares
+    module Memory
+      private
+
+      def memory_initialize
+      end
+
+      def memory_start(env)
+        Memory.memory_query_data = memory_parse_data
+      end
+
+      def memory_finish(env)
+        Memory.memory_query_data = memory_parse_data(Memory.memory_query_data)
+        save_report env, :memory,
+                    memory_short_form_data,
+                    memory_long_form_data
+      ensure
+        Memory.memory_query_data = nil
+      end
+
+      def memory_long_form_data
+        proc { Memory.memory_query_data }
+      end
+
+      alias_method memory_long_form_data memory_short_form_data
+
+      def self.memory_query_data
```

Comment:
Any reason this, and the following method, need to be class methods?  Seems like they would work just fine now as instance methods (less typing is also a plus).